### PR TITLE
Develop market

### PR DESCRIPTION
**Pull request by JoylimJY**
update Market and Trade

1. fetch and display topVolume, winner and loser tokens on market page.
2. redirection from market page to swap page when clicking on entries in the _Pairs_ table (very bottom).
3. render `<TradeHistoryTable />` with fetched data.

### CODE REVIEW

**Comment by ykchong45** [comment]
### Trade page price chart
For on-chain existing pairs, exchange rate is stored in db, chart can be plotted without calcuation.
For custom swapping pairs (which might not exist), rate is calculated by ratio of token price at corresponding timestamp. **_When either token does not have a transaction, calculation may not have a numerical result (NAN)_**

File: src/components/SwapComponent/index.js [hunk 1]
```diff
@@ -18,32 +18,26 @@ import styles from './styles.less';
...
+    library,
+    chainId,
   } = props;
   // 选择货币的弹窗
 
-  const { chainId } = useChainId();
-  const INITIAL_TOKEN_LIST = getTokens(chainId);
-
-  const coinList = getGlobalTokenList()
+  const { farmSetting: { INITIAL_ALLOWED_SLIPPAGE } } = useConstantLoader(props);
```
Comment (ykchong45) [inline src/components/SwapComponent/index.js]: Please remove this if possible, thanks! 🙏

File: src/pages/Market/Index.js [hunk 2]
```diff
@@ -344,28 +279,28 @@ const MarketIndex = props => {
 
             <div className={styles.statscontent}>
               <div className={styles.statsRow}>
-                <div className={styles.label}>1. WETH/</div>
-                <div className={styles.value}>XXX</div>
+                <div className={styles.label}>1. {winnersPairs[0]?.name}/</div>
+                <div className={styles.value}>{winnersPairs[0]?.price}</div>
```
Comment (ykchong45) [inline src/pages/Market/Index.js]: Percentage changes of token price should be displayed for _Winners_ and _Losers_. @JoylimJY @linja19

File: src/pages/Swap/Index.js [hunk 3]
```diff
@@ -20,40 +20,52 @@ import ExchangeTVChart from '@/components/ExchangeTVChart/ExchangeTVChart';
...
-  const [activeToken1, setActiveToken1] = useState(supportedTokens[3]);
-  const [activeToken0, setActiveToken0] = useState(supportedTokens[0]);
+const Swap = props => {
+  // const { farmSetting: { API_URL: apiUrlPrefix } } = useConstantLoader();
+  const { account, library, active } = useWeb3React()
+  
+  // test on polygon
+  const chainId = 137;
+
+  const [activeToken1, setActiveToken1] = useState({symbol: "USDC", address: "0x2791bca1f2de4661ed88a30c99a7a9449aa84174"});
```
Comment (ykchong45) [inline src/pages/Swap/Index.js]: Undesired hardcoded value found on this and the next line. Change this to something dependent on `chainId`.

File: src/pages/Swap/Index.js [hunk 4]
```diff
@@ -20,40 +20,52 @@ import ExchangeTVChart from '@/components/ExchangeTVChart/ExchangeTVChart';
...
 
   useEffect(() => {
-    library.on('block', (blockNum) => {
-      updateTxList();
-    })
-    return () => {
-      library.removeAllListeners('block');
+    if (active) {
+      library.on("block", () => {
+        // updateTxList()
```
Comment (ykchong45) [inline src/pages/Swap/Index.js:56]: why is this line commented out?

**Review by ykchong45 — CHANGES_REQUESTED** [review]
1. Still missing data for **Trending** section.
image
2. In the very bottom _Pairs table_, total liquidity column and price column does not seem to have correct parsings. 
3e442fdb01ffe476be735ef50ffcb2f
3. I might have missed how data is loaded in the NewPairs table (on the top). Are they real data from server?

File: src/pages/Swap/Index.js (hunk 4 above)
Comment (JoylimJY) [reply src/pages/Swap/Index.js:56]: because the variable doesn't seem necessary

File: src/pages/Market/Index.js [hunk 5]
```diff
@@ -27,7 +27,7 @@ const MarketIndex = props => {
         name: pairlist[i].token0.replace('Wrapped ', 'W') + '/' + pairlist[i].token1.replace('Wrapped ', 'W'),
         address0: pairlist[i].token0Address,
         address1: pairlist[i].token1Address,
-        price: '$' + (pairlist[i].token0Price / pairlist[i].token1Price).toFixed(2),
+        price: (pairlist[i].token0Price / pairlist[i].token1Price).toFixed(2),
         price_24h: '$' + parseFloat(pairlist[i].priceVariation).toFixed(4),
```
Comment (ykchong45) [inline src/pages/Market/Index.js:31]: Just to check if the unit of `priceVariation` is %?

**Comment by ykchong45** [comment]
image

**Review by ykchong45 — APPROVED** [review]

**Pull request merged**